### PR TITLE
Fix char() for different integer kinds

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lfortran/lfortran/issues/2042

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
This PR needs to be rebased after merging/completion of https://github.com/lfortran/lfortran/pull/2057.

**Comment by certik** [comment]
Can you send this PR to LPython as well? To make sure there is no issue there.

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 1]
```diff
@@ -1353,7 +1353,8 @@ LFORTRAN_API int _lfortran_str_ord_c(char* s)
 LFORTRAN_API char* _lfortran_str_chr(int val)
 {
     char* dest_char = (char*)malloc(2);
-    dest_char[0] = val;
+    uint8_t extended_ascii = (uint8_t)val;
+    dest_char[0] = extended_ascii;
```
Comment (Shaikh-Ubaid) [inline src/libasr/runtime/lfortran_intrinsics.c:1357]: I am not sure if this is a good change. We do not test this at the moment.

Comment (Shaikh-Ubaid) [reply src/libasr/runtime/lfortran_intrinsics.c:1357]: It seems to be more for precaution (if in case a negative value gets passed)

**Comment by Shaikh-Ubaid** [comment]
>Can you send this PR to LPython as well? To make sure there is no issue there.

Please find https://github.com/lcompilers/lpython/pull/2203.

**Comment by certik** [comment]
Awesome, thanks. Let's merge https://github.com/lfortran/lfortran/pull/2060 first, then rebase this to make sure nothing broke.

**Review by certik — APPROVED** [review]
I think this looks good, thank you @Shaikh-Ubaid.

**Comment by Shaikh-Ubaid** [comment]
I think the CI failure in `CI / Test LLVM 14 (pull_request)` is unrelated. Just running it should help.

**Pull request merged**